pkg/harness/tool/terraform/api/state.go

```diff
@@ -24,11 +26,49 @@ func OutputValueString(value interface{}) string {
 	return string(result)
 }
 
+func CloneMap(m map[string]any) map[string]any {
+	c, err := copystructure.Copy(m)
+	if err != nil {
+		return maps.Clone(m) // Return shallow copy if deep copy fails.
+	}
+
+	return c.(map[string]any)
+}
+
+func ExcludeSensitiveValues(values map[string]any, sensitiveValues map[string]any) map[string]any {
+	out := CloneMap(values)
```

Comment:
do you really need to clone the map on each level of recursion (it will make this a good bit less performant).  I'm guessing the only really risky part of doing it in place is destabilizing the `for k, v :=  range ...` but you could just put the keys in a new list and range over those perhaps?

Reply:
It should be possible to start with an empty map and only append non-sensitive values. thus avoiding direct copy.

Reply:
I have changed it to do it in place.

Reply:
@michaeljguarino PTAL